metakb/harvesters/moalmanac.py

```diff
@@ -0,0 +1,388 @@
+        """
+        Retrieve and store sources, variants, and assertions
+        records from MOAlmanac in composite and individual JSON files.
+
+        :return:'True' if successfully retreived, 'False' otherwise
+        :rtype: bool
+        """
+        try:
+            sources = self._harvest_sources()
+            variants = self._harvest_variants()
+            # genes = self._harvest_genes(variants)
+            assertions = self._harvest_assertions(variants)
+            self._create_json(assertions, sources, variants)
+            logger.info('MOAlamanc harvester was successful.')
+            return True
+        except:  # noqa: E722 # TODO: add details of exception error
+            logger.info('MOAlamanc harvester was not successful.')
+            return False
+
+    def _create_json(self, assertions, sources, variants):
+        """
+        Create a composite JSON file containing assertions,
+        sources, and variants
+        and individual JSON files for each level of MOA record.
+
+        :param: A list of MOA assertions
+        :param: A list of MOA sources
+        :param: A list of MOA variants
+        """
+        composite_dict = {
+            'assertions': assertions,
+            'sources': sources,
+            'variants': variants,
+        }
+
+        # Create composite json
+        moa_dir = PROJECT_ROOT / 'data' / 'moa'
+        moa_dir.mkdir(exist_ok=True, parents=True)
+        with open(f'{PROJECT_ROOT}/data/moa/moa_harvester.json',
+                  'w+') as f:
+            json.dump(composite_dict, f)
+            f.close()
+
+        # Create individual json for assertions, sources, variants
+        data = ['assertions', 'sources', 'variants']
+        for d in data:
+            with open(f'{PROJECT_ROOT}/data/moa/{d}.json', 'w+') as f:
+                f.write(json.dumps(composite_dict[d]))
+                f.close()
+
+    def _harvest_sources(self):
+        """
+        Harvest all MOA sources
+
+        :return: A list of sources
+        :rtype: list
+        """
+        sources_list = []
+        with requests_cache.disabled():
+            r = requests.get('https://moalmanac.org/api/sources')
+            sources = r.json()
+            for source in sources:
+                e = self._source_item(source)
+                sources_list.append(e)
+        return sources_list
+
+    def _harvest_variants(self):
+        """
+        Harvest all MOA variants
+
+        :return: A list of variants
+        :rtype: list
+        """
+        with requests_cache.disabled():
+            r = requests.get('https://moalmanac.org/api/attribute_definitions')
+            attr_def = r.json()
+            attr_def_idToname = self._get_attr_def(attr_def)
+
+        with requests_cache.disabled():
+            r = requests.get('https://moalmanac.org/api/feature_definitions')
+            feat_defs = r.json()
+            feat_def_nameToattr_def_id = self._get_feat_def(feat_defs)
```

Comment:
Same as above comment. 
```suggestion
            feat_def_name_to_attr_def_id = self._get_feat_def(feat_defs)
```